test/e2e/testsuites/azdiskschedulerextender_pod_scheduling_on_failover_multiple_pv_perf_tester.go

```diff
@@ -77,16 +79,21 @@ func (t *AzDiskSchedulerExtenderPodSchedulingOnFailoverMultiplePV) Run(client cl
 
 	time.Sleep(10 * time.Second)
 	for i := 0; i < 3; i++ {
-		podutil.DeleteAllPodsWithMatchingLabel(client, namespace, map[string]string{"app": "azuredisk-volume-tester"})
+		podutil.DeleteAllPodsWithMatchingLabel(client, namespace, map[string]string{"group": "delete-for-failover"})
 		for _, tStatefulSet := range tStatefulSets {
-			wg.Add(1)
 			go func(ss *resources.TestStatefulset) {
-				defer wg.Done()
 				defer ginkgo.GinkgoRecover()
-				ss.WaitForPodReady()
+				err := ss.WaitForPodReady()
+				errorsChan <- err
 			}(tStatefulSet)
 		}
-		wg.Wait()
+
+		for range tStatefulSets {
+			err := <-errorsChan
+			if err != nil {
+				framework.ExpectNoError(err)
```

Comment:
It would be a good idea to add an explanatory message to more readily identify the point of failure, e.g. `framework.ExpectNoError(err, "Failed waiting for StatefulSet pod failover.")`